db-service/lib/cqn2sql.js

```diff
@@ -696,36 +696,18 @@ class CQN2SQLRenderer {
    * @returns {string} The correct operator string
    */
   operator(x, i, xpr) {
-    // Convert into ANSI SQL NULL operators
-    if (x === '=' && xpr[i + 1]?.val === null) return 'is'
-    if (x === '!=' && xpr[i + 1]?.val === null) return 'is not'
+    if (x === '=')  return xpr[i + 1]?.val === null ? 'is'     : _not_null(xpr[i-1]) && _not_null(xpr[i+1]) ? '='  : this.is_
+    if (x === '!=') return xpr[i + 1]?.val === null ? 'is not' : _not_null(xpr[i-1]) && _not_null(xpr[i+1]) ? '<>' : this.is_not_
     return x
+    function _not_null(operand) {
+      if (!operand) return
+      operand = operand.element || operand.val
```

Comment:
```js
operand = operand.element || !(operand.val == null)
```
If `val` is `undefined` it would need to be something else. When the `val` is `0` or `''` it should should be not null. As we don't cache our prepared statements all instances of `{val:null}` will be checked and written into the `SQL`.

Reply:
Trying to understand... any specific reason for `!(operand.val == null)` vs `operand.val != null`?

Reply:
Btw: I took this `operand = operand.element || operand.val` from the original implementation:

```js
    left = left.element || left.val
    right = right.element || right.val
```

( So I thought at least )

Reply:
Mostly to be on the safe side. As I know for certain that `undefined == null` returns `true`, but was not sure wehter `undefined != null` would return `false`. Double checked it and they do behave the same.

Reply:
The original implementation was also wrong, because it did not consider the `0` and `''` cases.

Reply:
This PR shows how a small optimization like this can easily break things. Which is also why the original implementation was intentionally kept dumb.